apps/aechannel/src/aesc_utils.erl

```diff
@@ -463,24 +463,28 @@ check_root_hash_in_channel(Channel, PoI) ->
 %%%===================================================================
 
 process_solo_close(ChannelPubKey, FromPubKey, Nonce, Fee,
-                   Payload, PoI, Height, Trees) ->
-    process_solo_close_slash(ChannelPubKey, FromPubKey, Nonce, Fee,
-                             Payload, PoI, Height, Trees).
+                   Payload, PoI, Height, Trees, Env) ->
+    add_event(
+      process_solo_close_slash(ChannelPubKey, FromPubKey, Nonce, Fee,
+                               Payload, PoI, Height, Trees),
+      ChannelPubKey, Env).
 
 
 process_slash(ChannelPubKey, FromPubKey, Nonce, Fee,
-              Payload, PoI, Height, Trees) ->
-    process_solo_close_slash(ChannelPubKey, FromPubKey, Nonce, Fee,
-                             Payload, PoI, Height, Trees).
+              Payload, PoI, Height, Trees, Env) ->
+    add_event(
+      process_solo_close_slash(ChannelPubKey, FromPubKey, Nonce, Fee,
+                               Payload, PoI, Height, Trees),
+      ChannelPubKey, Env).
 
-process_solo_snapshot(ChannelPubKey, FromPubKey, Nonce, Fee, Payload, Trees) ->
+process_solo_snapshot(ChannelPubKey, FromPubKey, Nonce, Fee, Payload, Trees, Env) ->
     ChannelsTree0      = aec_trees:channels(Trees),
     Channel0 = aesc_state_tree:get(ChannelPubKey, ChannelsTree0),
     {ok, _SignedTx, PayloadTx} = deserialize_payload(Payload),
```

Comment:
Here you have the `SignedTx` and ignore it? And then do some work to get it in `add_event`? Or is it another transaction.

Reply:
At this level, I don't know, and don't care, what's in the payload. It is, in fact the offchain state, so a different tx. Since I'm making changes anyway, I'll change the variable name to reflect this.